src/command/expire/expire.c

```diff
@@ -609,9 +623,12 @@ removeExpiredBackup(InfoBackup *infoBackup)
         {
             LOG_INFO("remove expired backup %s", strPtr(strLstGet(backupList, backupIdx)));
 
-            storagePathRemoveP(
-                storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s", strPtr(strLstGet(backupList, backupIdx))),
-                .recurse = true);
+            // execute the real expiration and deletion
+            // only if the dry-run mode is disabled
+            if ( ! cfgOptionBool(cfgOptDryRun) )
+              storagePathRemoveP(
+                                 storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s", strPtr(strLstGet(backupList, backupIdx))),
+                                 .recurse = true);
         }
     }
 
```

Comment:
Now that infoBackup in memory is updated, we do not want to save it to disk if doing a dry-run. So look for the following line:
```
infoBackupSaveFile(
    infoBackup, storageRepoWrite(), INFO_BACKUP_PATH_FILE_STR, cipherType(cfgOptionStr(cfgOptRepoCipherType)),
    cfgOptionStr(cfgOptRepoCipherPass));
```
And wrap it in the conditional with a comment maybe like "Do not save the updated backup.info file if performing a dry-run".